src/analyses/apron/relationAnalysis.apron.ml

```diff
@@ -740,7 +740,12 @@ struct
         in
         ({ f } : Queries.ask) in
       let rel = RD.assert_inv dummyask rel e false (no_overflow ask e_orig) in (* assume *)
-      let rel = RD.keep_vars rel (List.map RV.local vars) in (* restrict *)
+      let rel =
+        if GobConfig.get_bool "ana.apron.strengthening" then
```

Comment:
I just realized we have a bit of an inconsistency with how our relation and Apron stuff are split. This analysis is also applied for our OCaml-implemented relational domains, but only `ApronDomain` implements the strengthening logic.
Given everything else (working with Apron variables and expressions), it might be possible to move the strengthening implementation up a bit such that it also works for others. Not that this would make a notable practical difference though.

Reply:
It might, once @DrMichaelPetter's additional relational domains make a debut....

Reply:
Maybe this can be extracted into an issue?